### DISCUSSION TITLE
Using the grid more than necessary....

### DISCUSSION THREAD

**Discussion by nickz98** [netzero-labs/netzero#14] [Ideas]
Hi,  Great app - thank you.
My system: 10Kw PV + Powerwall 2 + GSHP Heat pump (up to 18kw).  UK FIT scheme from 2011, meaning zero  benefit to exporting.
I get occasional peak demands for electric, for example today at 6.5Kw, for maybe 5-10mins.    At roughly midday on a reasonably sunny day my PV was pulling in 5KW and my powerwall was at 93% full.  Obviously / ideally my battery would be used instead of the grid for that shortfall of 1.5Kw.  However I think NETZERO may have my system set to mid-peak / mid-off peak...meaning the powerwall may decide to use the grid.   If there were netzero config options to be able to set the system to use either off peak or peak (not the options in between) then I think powerwall would have done that, i.e. even though it's a cheap time to import if I've said it's peak cost then the pw will use the battery instead.      I appreciate the calcs aren't straightforward, but I think that would work out better generally?    So my suggestion to be able to interfere / config so that I can only use say 2 of the pricing types (peak & off peak )?

**Comment by eJonny** [comment]
Are you leaving your mode set to Time Based Control? I've found the TBC pulls more from the grid than I would like. I use Netzero to change mode from TBC to Self Powered when I want my system only draw load from solar gen and my power walls.

**Reply by nickz98** [reply]
@eJonny , I think I need to do the same.   Trouble is knowing when I want to switch between the 2 etc.    I guess broadly speaking if it's sunny then I ideally don't want the grid being used whenever possible.    Are you able to share from your experience which times / automations you've set to switch between the 2?
Thanks

**Comment by zigam** [comment]
@nickz98 sounds like you're using the Agile Octopus integration?  There are no off/mid/peak periods in that case, every 30 min period has its own price and they change every day.  Powerwall makes decisions based on the future prices, estimated home usage, and estimated solar production.  The only goal in Time-Based Control is to maximize savings, so it will prioritize making sure you're not pulling from the grid at peak time.  When your goals are different (e.g. avoid pulling from the grid altogether) then Self Powered mode might make more sense, as @eJonny suggested.  Switching between those two modes is also an option, and you can use Netzero automation to schedule that.

**Comment by nickz98** [comment]
Thanks both.  Mine was set to TBC, as that's what I was using when I updated them manually (and have done for a week or 2).....until I found your app! (thank you!).
I had an example just after 6pm (prime charging with Octopus agile is from 4pm until 7pm.....so the most expensive time.)    I was short by around 0.2Kw (big money!), but it decided to pull the electric from the grid rather than the 100% charged pw. 
When configuring manually, I was able to set a zillion differnt time slots, and the 4 deak / not peaks etc.    Though I deliberatly only used peak and off peak - which give gave me control to peak (don;t use the grid unless I'm at 0% charge)   or off-peak (charge the battery if it's not full)....although it did sometimes hold back on that if it was sunny , therefore allowing the solar to do the charging and therefore not pull from the grid.  Clearly the pw was doing more forecasting etc than me saying I used it as a simple on / off switch.   

But still, for "peak" it would never charge from the grid whilst I had charge in the battery (unless my demand from the battery was over 5kw of course).

**Reply by zigam** [reply]
Unfortunately Powerwall behavior is a bit of a black box, so I'm not sure what the issue might be.  Other than these instances, is the Powerwall charging and discharging sensibly given the tariffs?

I know it can take the battery a few seconds to react to changes in demand; also, the charts are 5 minute averages, so they're hiding real peak values (what may seem like below 5kW usage on the chart could have been a higher instantaneous load).  But I'm not sure if this explains what you're seeing.
